underfs/b2/src/main/java/alluxio/underfs/b2/B2UnderFileSystem.java

```diff
@@ -0,0 +1,286 @@
+            USER_AGENT);
+
+    try {
+      B2AccountAuthorization accountAuthorization = client.getAccountAuthorization();
+      LOG.debug(accountAuthorization.toString());
+
+      B2Allowed allowed = accountAuthorization.getAllowed();
+      LOG.debug(allowed.toString());
+
+      String accountId = accountAuthorization.getAccountId();
+      List<String> capabilities = allowed.getCapabilities();
+
+      String bucketNameFromURI = UnderFileSystemUtils.getBucketName(alluxioURI);
+      LOG.debug("bucketname from uri: {}", bucketNameFromURI);
+
+      String bucketId;
+      String bucketName;
+      if (allowed.getBucketId() != null) {
+        bucketId = allowed.getBucketId();
+        bucketName = allowed.getBucketName();
+      } else {
+        B2Bucket bucket = client.getBucketOrNullByName(bucketNameFromURI);
+        LOG.debug(bucket.toString());
+        bucketId = bucket.getBucketId();
+        bucketName = bucket.getBucketName();
+      }
+
+      short bucketMode = B2Utils.translateBucketAcl(capabilities);
+
+      return new B2UnderFileSystem(alluxioURI, conf, client, bucketId, bucketName, bucketMode,
+          accountId);
+    } catch (B2Exception | NullPointerException e) {
+      LOG.error("Failed to instantiate B2UnderFileSystem client with key:{} and URI: {}",
+          conf.get(PropertyKey.B2_ACCESS_KEY), alluxioURI);
+      Throwables.propagateIfPossible(e, ServiceException.class);
+    }
+    return null;
+  }
+
+  protected B2UnderFileSystem(AlluxioURI alluxioURI, UnderFileSystemConfiguration conf,
+      B2StorageClient b2StorageClient, String bucketId, String bucketName, short bucketMode,
+      String accountId) {
+    super(alluxioURI, conf);
+    requireNonNull(alluxioURI, "alluxioURI is null");
+    requireNonNull(conf, "conf is null");
+    requireNonNull(b2StorageClient, "b2StorageClient is null");
+    requireNonNull(bucketId, "bucketId is null");
+    requireNonNull(bucketName, "bucketName is null");
+    requireNonNull(bucketMode, "bucketMode is null");
+    requireNonNull(accountId, "accountId is null");
+
+    LOG.debug("alluxioURI: {}", alluxioURI);
+    LOG.debug("buckedId: {}", bucketId);
+    LOG.debug("bucketName: {}", bucketName);
+    LOG.debug("bucketMode: {}", bucketMode);
+    LOG.debug("accountId: {}", accountId);
+
+    mB2StorageClient = b2StorageClient;
+    mBucketId = bucketId;
+    mBucketName = bucketName;
+    mBucketMode = bucketMode;
+    mAccountId = accountId;
+  }
+
+  @Override public boolean createEmptyObject(String key) {
+    return false;
+  }
+
+  @Override protected OutputStream createObject(String key) throws IOException {
+    return null;
+  }
+
+  @Override protected boolean copyObject(String src, String dst) throws IOException {
+    return false;
+  }
+
+  //
+  @Override protected boolean deleteObject(String key) throws IOException {
+    return false;
+  }
+
+  @Override protected ObjectPermissions getPermissions() {
+    LOG.debug("getPermissions");
```

Comment:
I don't think this log message is very helpful

Reply:
I removed this line